earth_enterprise/src/common/performancelogger.h

```diff
@@ -87,4 +101,10 @@ class BlockPerformanceLogger {
   BlockPerformanceLogger<PerformanceLogger> name(op, __VA_ARGS__)
 #define END_PERF_LOGGING(name) name.end()
 
+#define BEGIN_IO_LOGGING() // fill in as needed
+#define BEGIN_IO_LOGGING() // fill in as needed
+#define BEGIN_THREAD_LOGGING() // fill in as needed
+#define BEGIN_THREAD_LOGGING() // fill in as needed
```

Comment:
Perhaps create a third macro which creates the envelope?

Like

 `#define CREATE_IO_ENVELOPE(envlp, ...)`